hydrolib/core/io/polyfile/parser.py

```diff
@@ -0,0 +1,554 @@
+            self._feed_line[self._state](line)
+
+    def _handle_parse_name(self, line: str) -> None:
+        self._current_block.name = Parser._convert_to_name(line)
+        self._state = StateType.PARSED_NAME
+
+    def _handle_new_description(self, line: str) -> None:
+        comment = Parser._convert_to_comment(line)
+        self._current_block.description = [
+            comment,
+        ]
+        self._state = StateType.PARSED_DESCRIPTION
+
+    def _handle_next_description(self, line: str) -> None:
+        comment = Parser._convert_to_comment(line)
+        self._current_block.description.append(comment)  # type: ignore
+
+    def _handle_empty_line(self) -> None:
+        if self._state != StateType.INVALID_STATE:
+            self._current_block.empty_lines.append(self.line)
+
+    def _log_ws_warning(self, line: str) -> None:
+        if line[0] != " ":
+            return
+
+        end_column = len(line) - len(line.lstrip()) - 1
+        self._current_block.ws_warnings.append(
+            ParseMsg(
+                line=(self.line, self.line),
+                column=(0, end_column),
+                reason="White space at the start of the line is ignored.",
+            )
+        )
+
+    def _handle_new_error(self, reason: str) -> None:
+        self._error_builder.start_invalid_block(
+            self._current_block.start_line, self.line, reason
+        )
+        self._state = StateType.INVALID_STATE
+
+    def _handle_parse_msg(self, msg: ParseMsg) -> None:
+        msg.notify_as_warning(self._file_path)
+
+    @staticmethod
+    def _is_empty_line(line: str) -> bool:
+        return len(line.strip()) == 0
+
+    @staticmethod
+    def _is_name(line: str) -> bool:
+        stripped = line.strip()
+        return len(stripped) >= 1 and line[0] != "*" and not (" " in stripped)
+
+    @staticmethod
+    def _convert_to_name(line: str) -> str:
+        return line.strip()
+
+    @staticmethod
+    def _is_comment(line: str) -> bool:
+        stripped = line.strip()
+        return len(stripped) >= 1 and stripped[0] == "*"
+
+    @staticmethod
+    def _convert_to_comment(line: str) -> str:
+        return line.strip()[1:]
+
+    @staticmethod
+    def _convert_to_dimensions(line: str) -> Optional[Tuple[int, int]]:
+        stripped = line.strip()
+        elems = stripped.split()
+
+        if len(elems) != 2:
+            return None
+
+        try:
+            n_rows = int(elems[0])
+            n_cols = int(elems[1])
+
+            if n_rows <= 0 or n_cols <= 0:
+                return None
+
+            return (n_rows, n_cols)
+        except ValueError:
+            return None
+
+    @staticmethod
+    def _convert_to_point(
+        line: str, expected_n_points: int, has_z: bool
+    ) -> Optional[Point]:
+        stripped = line.strip()
+        elems = stripped.split()
+
+        if len(elems) != expected_n_points:
+            return None
+
+        try:
+            values = list(float(x) for x in elems)
```

Comment:
I personally would like to keep the explicit conversion here, because I specifically want to catch an invalid conversion from strings to ints. Since I know this will be thrown as a `ValueError` I can filter the `except` on that. The `ValidationError` would be more generic, and might indicate other problems. 

I did not know that Pydantic will do that magically though, which is pretty nice (: